.github/workflows/mega-linter-for-runner.yml

```
@@ -41,133 +123,228 @@ jobs:
+
+        ##
+        # These variables will override project configurations, including those
+        # set in the `.mega-linter.yml` file. This is useful for creating
+        # conditional exceptions during GitHub workflows, for example, when
+        # linting a production deployment.
+        #
+        # https://megalinter.io/latest/config-file/
+        #
         env:
-          DEFAULT_WORKSPACE: mega-linter-runner # ml workflow addition
 
-          # Validates all source when push on main, else just the git diff with
-          # main. Override with true if you always want to lint all sources
+          ##
+          # By default, this validates full codebase whenever a push to the
+          # default branch occurs. Otherwise, only the `git diff` files will
+          # be linted for better efficiency.
           #
-          # To validate the entire codebase, set to:
-          # VALIDATE_ALL_CODEBASE: true
+          # This can be overridden to always lint the codebase with:
+          # 
+          #   VALIDATE_ALL_CODEBASE: true
+          # 
+          # To validate only `git diff`` with the default branch on push:
           #
-          # To validate only diff with main, set to:
           # VALIDATE_ALL_CODEBASE: >-
           #   ${{
           #     github.event_name == 'push' &&
-          #     contains(fromJSON('["refs/heads/main", "refs/heads/master"]'), github.ref)
-          #   }}
-          VALIDATE_ALL_CODEBASE: true # ml workflow change
-          # >-
-          #   ${{
-          #     github.event_name == 'push' &&
-          #     contains(fromJSON('["refs/heads/main", "refs/heads/master"]'), github.ref)
+          #     (
+          #       github.ref == 'refs/heads/main' ||
+          #       github.ref == 'refs/heads/master'
+          #     )
           #   }}
+          #
+          VALIDATE_ALL_CODEBASE: true
 
+          ##
+          # This token is automatically created on the GitHub server and you
+          # should not modify this line.
+          #
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-          # ADD YOUR CUSTOM ENV VARIABLES HERE OR DEFINE THEM IN A FILE
-          # .mega-linter.yml AT THE ROOT OF YOUR REPOSITORY
+          ##
+          # Add any other variables you wish to override from `.mega-linter.yml`
+          # below this line.
+          #
 
-          # Uncomment to disable copy-paste and spell checks
-          # DISABLE: COPYPASTE,SPELL
+          DEFAULT_WORKSPACE: mega-linter-runner
 
-      # Upload MegaLinter artifacts
+      ##
+      # Upload MegaLinter artifacts.
+      #
+      # This stores each report and log from tests for your reference on the
+      # GitHub Action report.
+      #
+      # More info at:
+      # https://github.com/actions/upload-artifact
+      # https://megalinter.io/latest/reporters/
+      #
       - name: Archive production artifacts
+        if: success() || failure()
         uses: actions/upload-artifact@v3
```

```suggestion
        uses: actions/upload-artifact@v3
        if: success() || failure()
```
To match mega-linter-runner/generators/mega-linter/templates/mega-linter.yml